### ISSUE TITLE
Ability to index a DataTree with a PurePosixPath 

### ISSUE THREAD

**Issue by eschalkargans** [pydata/xarray#9253]
### Is your feature request related to a problem?

When working with DataTrees, that are tree-like structure, it is common to have to build paths to access the data contained inside of it. It is then natural to envision `pathlib` as an help to build such paths, rather than relying on strings. DataTree already uses its NodePath structure internally, that inherit from PurePosixPath

On can already do this:

```python
path = PurePosixPath("/a/b/c/")
xdt: DataTree
xdt[str(path)]
```

But having to add this `str` call adds noise to the code.

### Describe the solution you'd like

```python
xdt: DataTree
xdt[PurePosixPath("/a/b") / "c"]
```

### Describe alternatives you've considered

Calling `str` to an existing `PurePosixPath` is the most natural way: use pathlib to build a path and downgrade to a string accepter by the DataTree indexing logic

**Comment by TomNicholas** [comment]
I've thought about doing this before, and it wouldn't be hard. I think we would welcome a PR to add this. 

The main question is how flexible do we want to be in accepting subclasses? I think your suggestion of accepting either PurePosixPath or NodePath (which we use internally) is nice, but we probably don't want to accept a concrete PosixPath, so that the tree structure can't be confused with the directory structure on-disk.

(Also I know we need to get back to your PRs on the original DataTree repo @eschalkargans !!)

**Comment by eschalkargans** [comment]
>  I think your suggestion of accepting either PurePosixPath or NodePath (which we use internally) is nice, but we probably don't want to accept a concrete PosixPath, so that the tree structure can't be confused with the directory structure on-disk.

Yes definitely, the `PurePosixPath` has the strong meaning of "this only a Path, not filesystem-related"!
